### ISSUE TITLE
About how to finetune RoBERTa on SQuAD 

### ISSUE THREAD

**Issue by kugwzk** [pytorch/fairseq#933]
Thanks for the RoBERTa.
I want to fine tune the model on SQuAD dataset, but I couldn't find the detail script and some codes for that. Do you will give something about that?

**Comment by schinger** [comment]
same here

**Comment by myleott** [comment]
Yes, we will provide scripts and a README for SQuAD finetuning soon.

**Comment by schinger** [comment]
Does SQuAD finetuning scripts will be compatible with huggingface/pytorch-transformers ?

**Comment by Damcy** [comment]
any update?

**Comment by myleott** [comment]
Sorry, we still need to refactor and cleanup the code. We will release it, but in the meantime you can try using the SQuAD finetuning for RoBERTa in pytorch-transformers.

**Comment by ecchochan** [comment]
I followed the hyper parameters in the roBERTa paper, using fairseq to train roBERTa on SQuAD task.


Input representation: `<s> Passage here. </s> Q: Question here? </s>`

```
Batch Size:           48 
Max Epochs:           2
Warmup ratio:         0.06
Number of Steps:      5430
Warmup Steps:         326
Learning Rate:        1.5e-5 
Weight Decay:         0.01 
Learning Rate Decay:  Linear
Adam ǫ:               1e-6
```

I got the following evaluation:
```json
{
  "best_exact": 85.21014065526826,
  "best_exact_thresh": -1.6142578125,
  "best_f1": 88.29940280085873,
  "best_f1_thresh": -1.572265625
}
```

Some distance from `89.4` in the paper :(

Did I miss something? Thank you for the great work!

**Comment by niazarak** [comment]
@ecchochan
> Input representation: `<s> Passage here. </s> Q: Question here? </s>`

Isn't it supposed to have two separators between question and passage?

**Comment by OlegPlatonov** [comment]
Hi! Thanks for all the RoBERTa usage examples. I wonder if we can expect SQuAD finetuning example anytime soon?
I've been experimenting with adding different intermediate finetuning tasks to RoBERTa and I'm seeing some interesting results on GLUE. It would be great if I could test my approach on SQuAD as well. 
By the way, GLUE finetuning example is awesome!

**Comment by myleott** [comment]
@yinhanliu

**Comment by motefly** [comment]
Any update?

**Comment by zhiheng-huang** [comment]
This is one of the most important benchmark for question answering. The reproduction of the SOTA number is very useful to help the community to further improve accuracy. Would be appreciated if you can make the fine-tuning script available soon. Thanks!

**Comment by pisalore** [comment]
bump

**Comment by BaohaoLiao** [comment]
Hi @myleott @yinhanliu, are you still planning to release the finetuning and evaluation codes for SQuAD

**Comment by gabinguo** [comment]
bump

**Comment by Avo-k** [comment]
bump
